develop/getting-started/using-the-ide.md

```diff
@@ -0,0 +1,246 @@
+
+```java
+// private static final int PROTECTION_BOOTS = 2;
+private static final int PROTECTION_LEGGINGS = 5;
+// private static final int PROTECTION_CHESTPLATE = 6;
+private static final int PROTECTION_HELMET = 1;
+```
+
+```java
+/*
+ModItems.initialize();
+ModSounds.initializeSounds();
+ModParticles.initialize();
+*/
+
+private static int secondsToTicks(float seconds) {
+    return (int) (seconds * 20 /*+ 69*/);
+}
+```
+
+### Region comments
+
+In IntelliJ, right next to the line numbers, you can have small [+] and [-] icons. Those can be used to temporarily collapse methods, if-statements, classes and many other things
+if you are not actively working on them. To create a custom block which can be collapsed, use the `region` and `endregion` comments.
+
+```java
+// region collapse block name
+    ModBlocks.initialize();
+    ModBlockEntities.registerBlockEntityTypes();
+    ModItems.initialize();
+    ModSounds.initializeSounds();
+    ModParticles.initialize();
+// endregion
+```
+
+![Region collapse](/assets/develop/getting-started/using-the-ide/comments_02.png)
+
+> [!WARNING]
+> This feature may not be supported in other editors and IDEs.
+> If you notice that you are using too many of them, consider refactoring your code to make it more readable!
+
+### TODO and FIXME notes
+
+When working on code, it can come in handy to leave notes, on what still needs to be taken care of. Sometimes you may also spot
+a potential issue in the code, but you don't want to stop focusing on the current problem. In this case, use the
+`TODO` or `FIXME` comments.
+
+![TODO and FIXME comments](/assets/develop/getting-started/using-the-ide/comments_03.png)
+
+IntelliJ will keep track of them in the `TODO` window and may notify you, if you commit code,
+which uses those type of comments.
+
+![TODO and FIXME comments](/assets/develop/getting-started/using-the-ide/comments_04.png)
+
+![Commit with TODO](/assets/develop/getting-started/using-the-ide/comments_05.png)
+
+> [!WARNING]
+> This feature may not be supported in other editors and IDEs.
+
+### Javadocs
+
+A great way of documenting your code is the usage of JavaDoc. JavaDocs not only provide useful information for implementation of methods and classes, but are also
+deeply integrated into IntelliJ.
+
+When hovering over method or class names, which have JavaDoc comments added to them, they will display this information in their information window.
+
+![JavaDoc](/assets/develop/getting-started/using-the-ide/comments_06.png)
+
+To get started, simply write `/**` above the method or class definition and press enter. IntelliJ will automatically generate lines for the return value
+and the parameters but you can change them however you want. There are many custom functionalities available and you can also use HTML if needed.
+
+Minecraft's `ScreenHandler` class has some examples. To toggle the render view, use the pen button near the line numbers.
```

Comment:
Should note that any comments within minecraft's source code are from mappings, such as yarn.